src/module.c

```diff
@@ -560,17 +579,19 @@ int ReplySeriesRange(RedisModuleCtx *ctx, Series *series, api_timestamp_t start_
             RedisModule_ReplyWithDouble(ctx, sample.value);
             arraylen++;
         } while (SeriesIteratorGetNext(&iterator, &sample) == CR_OK &&
-                    (maxResults == -1 || arraylen < maxResults));
+                    (maxResults == -1 || arraylen < maxResults));            
     } else {
         bool firstSample = TRUE;
         do {
-            if (sample.timestamp >= last_agg_timestamp + time_delta) {
+            if ((iterator.reverse == NO_OPT && sample.timestamp >= last_agg_timestamp + time_delta) ||
+                (iterator.reverse == REVERSE && sample.timestamp <= last_agg_timestamp - time_delta)) {
                 if (firstSample == FALSE) {
                     ReplyWithAggValue(ctx, last_agg_timestamp, aggObject, context);
                     arraylen++;
                 }
-
-                last_agg_timestamp = sample.timestamp - ((sample.timestamp - last_agg_timestamp) % time_delta);
+                last_agg_timestamp = iterator.reverse == NO_OPT ?
+                    sample.timestamp - ((sample.timestamp - last_agg_timestamp) % time_delta) :
+                    sample.timestamp + ((last_agg_timestamp - sample.timestamp) % time_delta);
```

Comment:
@ashtul im not sure about this, IIUC this means that in reverse range we align the bucket to the MAX timestamp in the bucket, while in regular order we align to MIN.
Why not keep the buckets the same?

Reply:
This will change. It was part of the behavior change we reverted.